firedrake/dmplex.pyx

```diff
@@ -825,18 +848,15 @@ def get_facets_by_class(PETSc.DM plex, label):
                                   "op2_non_core",
                                   "op2_exec_halo",
                                   "op2_non_exec_halo"]):
+        CHKERR(DMPlexGetLabel(plex.dm, op2class, &lbl_class))
         nclass = plex.getStratumSize(op2class, 1)
         if nclass > 0:
-            class_is = plex.getStratumIS(op2class, 1)
-            CHKERR(ISGetIndices(class_is.iset, &indices))
-            for ci in range(nclass):
-                if fStart <= indices[ci] < fEnd:
-                    CHKERR(DMLabelHasPoint(lbl_facets, indices[ci],
-                                           &has_point))
-                    if has_point:
-                        facets[fi] = indices[ci]
-                        fi += 1
-            CHKERR(ISRestoreIndices(class_is.iset, &indices))
+            for f in ordering:
```

Comment:
It turns out cython does not appear to be smart enough to generate fast code for iterating over a numpy array, I will push a fix.